### PR TITLE
Improve time to convert binning behavior, add autobinning, expose in API

### LINKED ISSUE #4990: Improve funnel time to convert binning behavior
## Is your feature request related to a problem?

#4947 added a new query to the backend – time to convert. It works well, just one point is kind of rough: bin width and count.

## Describe the solution you'd like

Currently the number of bins is hardcoded. Ideally we should have two options:
- return a user-specified number of bins,
- otherwise infer a reasonable number of bins using some algorithm (auto option).

@alexkim205 [suggested](https://github.com/PostHog/posthog/issues/4855#issuecomment-872423877) the [Freedman-Diaconis rule](https://en.wikipedia.org/wiki/Freedman%E2%80%93Diaconis_rule) for determining bin width (and interval based on that), but unfortunately from my testing its formula would not be the most practical – it's great for keeping a pretty constant "smoothness" of the histogram esp. with normal distribution, but we face some additional constraints that make it less than ideal:
- we can't have histograms too dense visually in the UI, from my testing a practical limit is 60 bins (above that it gets too dense),
- our data generally doesn't follow much of a normal distribution.

Clamping may help though, or maybe a different formula may work better outright – I'll be testing this.

## Additional context

Issue pertaining to the experience of time to convert: #4855

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

This resolves #4990 by implementing a way to specify the number of bins by the user via `filter.bin_count` + adding default autobinning behavior (very very vaguely inspired by the [Freedman-Diaconis rule](https://en.wikipedia.org/wiki/Freedman%E2%80%93Diaconis_rule) – in that the number of bins is proportional to the cube root of the number of samples).

Practical details:
- an auto number of bins is clamped between 3 and 60,
- a custom number of bins is clamped between 1 and 90 (we trust the user a bit more so they get some leeway).

That is because the UI would be pretty impractical outside of these ranges.

Also exposes time to convert in the API (using `filter.funnel_viz_type`).

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests

### CODE REVIEW

File: ee/clickhouse/views/test/test_clickhouse_insights.py [hunk 1]
```diff
@@ -84,3 +82,45 @@ def test_funnel_trends_basic_post(self):
...
+
+        _create_event(event="step one", distinct_id="user b", team=self.team, timestamp="2021-06-09 13:00:00")
+        _create_event(event="step two", distinct_id="user b", team=self.team, timestamp="2021-06-09 13:37:00")
+        # Converted from 0 to 1 in 2200 s
+
+        _create_event(event="step one", distinct_id="user c", team=self.team, timestamp="2021-06-11 07:00:00")
+        _create_event(event="step two", distinct_id="user c", team=self.team, timestamp="2021-06-12 06:00:00")
+        # Converted from 0 to 1 in 82_800 s
+
+        with self.settings(DEBUG=1):
```
Comment (neilkakkar) [inline ee/clickhouse/views/test/test_clickhouse_insights.py:104]: Why was `DEBUG=1` necessary here? O.o

Comment (Twixes) [reply ee/clickhouse/views/test/test_clickhouse_insights.py]: A ha. It was not after I quickly resolved a 500 using it. Removed.

**Review by neilkakkar — APPROVED** [review]
LGTM, comments inside the query much appreciated, took me a while to get my head around it. But really glad it's all happening in the CTEs, so the main query remains as clean as possible. 👍

File: ee/clickhouse/views/test/test_clickhouse_insights.py (hunk 1 above)
Comment (neilkakkar) [reply ee/clickhouse/views/test/test_clickhouse_insights.py]: oooh, nice, does `DEBUG` print extra? (I usually go to the exception catcher and print there lol, this seems nicer)

Comment (Twixes) [reply ee/clickhouse/views/test/test_clickhouse_insights.py]: Yup, pretty good way to see the whole stack trace

**Pull request merged**